hibernate-core/src/test/java/org/hibernate/test/distinct/SelectDistinctHqlTest.java

```diff
@@ -108,6 +108,23 @@ public void test() {
 		} );
 	}
 
+	@Test
+	@TestForIssue( jiraKey = "HHH-11726" )
+	public void testDistinctPassThrough() {
+		// This should be returning false ref. HHH-11726
+		doInHibernate( this::sessionFactory, session -> {
+			sqlStatementInterceptor.getSqlQueries().clear();
+			List<Person> persons = session.createQuery(
+					"select distinct p from Person p left join fetch p.phones ")
+					.setHint(QueryHints.HINT_PASS_DISTINCT_THROUGH, true)
+					.setMaxResults(5)
+					.getResultList();
+			assertEquals(1, persons.size());
+			String sqlQuery = sqlStatementInterceptor.getSqlQueries().getLast();
+			assertTrue(sqlQuery.contains(" distinct "));
```

Comment:
if (you omit or set HINT_PASS_DISTINCT_THROUGH == true) then you'll see DISTINCT in the SQL query.

if (HINT_PASS_DISTINCT_THROUGH == false) then you will not see DISTINCT in the SQL query.